src/hotspot/share/adlc/adlc.hpp

```diff
@@ -108,4 +108,8 @@ typedef unsigned int uintptr_t;
 // it everywhere it needs to be available.
 extern ArchDesc* globalAD;
 
-#endif // SHARE_VM_ADLC_ADLC_HPP
+// Performs snprintf and asserts the result is non-negative (so there was not
+// an encoding error) and that the output was not truncated.
+extern int snprintf_checked(char* buf, size_t len, const char* fmt, ...);
+
+#endif // SHARE_ADLC_ADLC_HPP
```

Comment:
This shouldn't have come across with the patch, please also add the newline

```suggestion
```

Reply:
The forward declaration of `snprintf_checked` is indeed required, otherwise bad things happen [1]

I've submitted a new commit that replaces `SHARE_ADLC_ADLC_HPP` with `SHARE_VM_ADLC_ADLC_HPP` and adds the extra newline.


[1]
```
ERROR: Build failed for target 'images' in configuration 'linux-x86_64-normal-server-release' (exit code 2) 

=== Output from failing command(s) repeated here ===
* For target hotspot_variant-server_tools_adlc_objs_adlparse.o:
/home/antonio/REPOS/OPENJDK/jdk11u-dev/src/hotspot/share/adlc/adlparse.cpp: In member function 'void ADLParser::instr_parse()':
/home/antonio/REPOS/OPENJDK/jdk11u-dev/src/hotspot/share/adlc/adlparse.cpp:215:11: error: 'snprintf_checked' was not declared in this scope
  215 |           snprintf_checked(buf, buf_size, "%s_%d", instr->_ident, match_rules_cnt++);
      |           ^~~~~~~~~~~~~~~~
/home/antonio/REPOS/OPENJDK/jdk11u-dev/src/hotspot/share/adlc/adlparse.cpp: In member function 'void ADLParser::ins_encode_parse_block(InstructForm&)':
/home/antonio/REPOS/OPENJDK/jdk11u-dev/src/hotspot/share/adlc/adlparse.cpp:2864:3: error: 'snprintf_checked' was not declared in this scope
 2864 |   snprintf_checked(ec_name, ec_name_size, "%s%s", prefix, inst._ident);
      |   ^~~~~~~~~~~~~~~~
/home/antonio/REPOS/OPENJDK/jdk11u-dev/src/hotspot/share/adlc/adlparse.cpp: In member function 'void ADLParser::constant_parse(InstructForm&)':
/home/antonio/REPOS/OPENJDK/jdk11u-dev/src/hotspot/share/adlc/adlparse.cpp:3336:3: error: 'snprintf_checked' was not declared in this scope
 3336 |   snprintf_checked(ec_name, ec_name_size, "%s%s", prefix, inst._ident);
      |   ^~~~~~~~~~~~~~~~
/home/antonio/REPOS/OPENJDK/jdk11u-dev/src/hotspot/share/adlc/adlparse.cpp: In member function 'char* ADLParser::get_ident_or_literal_constant(const char*)':
/home/antonio/REPOS/OPENJDK/jdk11u-dev/src/hotspot/share/adlc/adlparse.cpp:4657:7: error: 'snprintf_checked' was not declared in this scope
 4657 |       snprintf_checked(buf, buf_size, "(%s)", param);
   ... (rest of output omitted)
```